### PR TITLE
ci: cache dependencies

### PR DESCRIPTION
**Pull request by mvadari**
## High Level Overview of Change

This PR adds caching of node_modules to testing. This shaves about a minute from the time to run CI.

### Context of Change

Everyone likes speedy CI

### Type of Change

- [x] Tests (You added tests for code that already exists, or your new feature included in this PR)

## Test Plan

Works in CI.

### CODE REVIEW

**Review by justinr1234 — APPROVED** [review]

File: .github/workflows/nodejs.yml [hunk 1]
```diff
@@ -18,37 +18,79 @@ jobs:
...
+      - uses: actions/checkout@v3
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+
+      - name: Setup npm version 8
+        id: npm8
+        run: |
+          npm i -g npm@7 --registry=https://registry.npmjs.org
```
Comment (JST5000) [inline .github/workflows/nodejs.yml:71]: This says it's setting up version 8 then installs version 7

File: .github/workflows/nodejs.yml [hunk 2]
```diff
@@ -97,14 +160,35 @@ jobs:
...
```
Comment (JST5000) [inline .github/workflows/nodejs.yml:172]: ditto

File: .github/workflows/nodejs.yml [hunk 3]
```diff
@@ -97,14 +160,35 @@ jobs:
...
+
+      - name: Cache node modules
+        id: cache-nodemodules
+        uses: actions/cache@v2
```
Comment (JST5000) [inline .github/workflows/nodejs.yml:176]: This is using `actions/cache@v2`, while the other additions are using `@v3`, is that intentional?

**Review by JST5000 — APPROVED** [review]
LGTM

**Pull request merged**